docs/source/advanced_usage.rst

```diff
@@ -32,6 +32,84 @@ locally):
 - QVM URL: ``http://127.0.0.1:5000``
 - quilc URL: ``tcp://127.0.0.1:5555``
 
+Concurrency
+~~~~~~~~~~~
+
+Using pyQuil for concurrent programming is as simple as calling :py:func:`~pyquil.get_qc()` from within a given context
+(be it a thread or process), then using the returned :py:class:`~pyquil.api.QuantumComputer` as usual within that
+context. While :py:class:`~pyquil.api.QuantumComputer` objects as a whole are not safe to share between contexts (due to
+state related to currently-running compilation or execution requests), some information they use is. Information related
+to client configuration (:py:class:`~pyquil.api.QCSClientConfiguration`) and QPU auth
+(:py:class:`~pyquil.api.EngagementManager`) can be safely extracted and shared among
+:py:class:`~pyquil.api.QuantumComputer` instances, as shown below, to save your code from redundant disk reads and
+auth-related HTTP requests.
+
+Using Multithreading
+--------------------
+
+.. code:: python
+
+    from multiprocessing.pool import ThreadPool
+
+    from pyquil import get_qc
+    from pyquil.api import EngagementManager, QCSClientConfiguration
+    from pyquil.quil import Program
```

Comment:
`from pyquil import Program`

Reply:
Nice. This is from auto-imports, but your suggestion is best for an example.

Reply:
Done.